artichoke-backend/src/extn/core/string/mod.rs

```diff
@@ -5,6 +5,7 @@ pub mod trampoline;
 pub struct String;
 
 #[cfg(test)]
+#[cfg(feature = "core-regexp")]
```

Comment:
Note to self: This feature dependency is because `String#[]` is implemented in pure Ruby and depends on `Regexp`.

https://github.com/artichoke/artichoke/blob/44b60ed9a8f3922cfac6c8e07ecc343e3091bc3e/artichoke-backend/src/extn/core/string/string.rb#L184-L197